File: _includes/sort-benchmark/algorithms/strand.rs
fn strand_sort(a: &mut [usize]) {
    let mut input = a.to_vec();
    let mut output = Vec::new();
    while !input.is_empty() {
        let mut strand = Vec::new();
        let mut rest = Vec::new();
        for value in input {
            if strand.last().map_or(true, |last| *last <= value) {
                strand.push(value);
            } else {
                rest.push(value);
            }
        }
        output = merge_values(&output, &strand);
        input = rest;
    }
    a.copy_from_slice(&output);
}

File: _includes/sort-benchmark/algorithms/strand.swift
func strand_sort(_ a: inout [Int]) {
    a.withUnsafeMutableBufferPointer { strand_sort($0) }
}

func strand_sort(_ a: UnsafeMutableBufferPointer<Int>) {
    var input = Array(a)
    var output = [Int]()
    while !input.isEmpty {
        var strand = [Int]()
        var rest = [Int]()
        for value in input {
            if strand.last.map({ $0 <= value }) ?? true {
                strand.append(value)
            } else {
                rest.append(value)
            }
        }
        output = merge_values(output, strand)
        input = rest
    }
    for i in 0..<a.count {
        a[i] = output[i]
    }
}
